src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -812,6 +927,169 @@ public static void ActionDoesNotExplicitlySpecifyContentTypes(this ILogger logge
             }
         }
 
+        public static void RegisteredModelBinderProviders(this ILogger logger, IModelBinderProvider[] providers)
+        {
+            _registeredModelBinderProviders(logger, providers, null);
+        }
+
+        public static void FoundNoValueOnRequest(this ILogger logger, ModelBindingContext bindingContext)
+        {
+            if (!logger.IsEnabled(LogLevel.Debug))
+            {
+                return;
+            }
+
+            var modelMetadata = bindingContext.ModelMetadata;
+            var isProperty = modelMetadata.ContainerType != null;
+            var modelName = GetModelName(bindingContext);
+
+            if (isProperty)
+            {
+                _foundNoValueForPropertyOnRequest(
+                    logger,
+                    modelName,
+                    modelMetadata.PropertyName,
+                    bindingContext.ModelType,
+                    modelMetadata.ContainerType,
+                    null);
+            }
+            else
+            {
+                _foundNoValueForParameterOnRequest(
+                    logger,
+                    modelName,
+                    modelMetadata.PropertyName,
+                    bindingContext.ModelType,
+                    null);
+            }
+        }
+
+        public static void NoPublicSettableProperties(this ILogger logger, ModelBindingContext bindingContext)
+        {
+            _noPublicSettableProperties(logger, GetModelName(bindingContext), bindingContext.ModelType, null);
+        }
+
+        public static void CannotBindToComplexType(this ILogger logger, ModelBindingContext bindingContext)
+        {
+            _cannotBindToComplexType(logger, bindingContext.ModelType, null);
+        }
+
+        public static void CannotBindToFilesCollectionDueToInvalidContentType(this ILogger logger, ModelBindingContext bindingContext)
+        {
+            _cannotBindToFilesCollectionDueToInvalidContentType(logger, GetModelName(bindingContext), bindingContext.ModelType, null);
```

Comment:
Could we include the actual content type in the log? And this would be UnsupportedContentType (not invalid)